packages/react-components/react-table/src/components/DataGridRow/DataGridRow.types.ts

```diff
@@ -12,7 +12,10 @@ export type DataGridRowSlots = TableRowSlots & {
   selectionCell?: Slot<typeof TableSelectionCell>;
 };
 
-export type CellRenderFunction<TItem = unknown> = (column: TableColumnDefinition<TItem>) => React.ReactNode;
+export type CellRenderFunction<TItem = unknown> = (
+  column: TableColumnDefinition<TItem>,
+  state: DataGridRowState,
```

Comment:
Actually here I rather meant that we could probably expose the `tableState` in these render functions (or a subset). Exposing `DataGridRowState` doesn't make too much sense here because then the user has access to things like element refs and internals from the component

Reply:
the table features are spread in the `dataGridContextValue` with other useful values. It feels like this data grid context is a good candidate to be passed as a render prop to its children. WDYT?